test/unit/system_uri_test.py

```diff
@@ -61,6 +61,30 @@ def test_translate_obs_uri_not_found(mock_request_get, self):
         uri.runtime_config = self.runtime_config
         assert uri.translate()
 
+    @raises(KiwiUriOpenError)
+    @patch('kiwi.system.uri.Defaults.is_buildservice_worker')
+    def test_translate_obs_uri_in_private_buildservice(
+            mock_buildservice, self
+    ):
+        mock_buildservice.return_value = True
+        self.runtime_config.is_obs_public = mock.Mock(
+            return_value=False
+        )
+        self.runtime_config.get_obs_download_server_url = mock.Mock(
+            return_value='obs_server'
+        )
```

Comment:
ok but coverage said the code was not reached, maybe because you are raising KiwiUriOpenError in a try block which raises again a KiwiUriOpenError... ah this is confusing

why do we check the public state ? I think it would be enough to log a warning that we can't verify the location. imho it would be possible somebody wants to add an obs repo as imageonly but build in the buildservice